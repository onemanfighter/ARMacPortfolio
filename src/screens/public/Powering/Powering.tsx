import { Box, Progress } from '@chakra-ui/react';
import { MacIcon } from '@assets';
import { useEffect, useState } from 'react';
import { processStore } from '@processStore';
import { powerSelector, useShallow } from '@processStore/selector';

const Powering = () => {
  const [value, setValue] = useState<number>(0);
  const { turnOn } = processStore(useShallow(powerSelector));

  useEffect(() => {
    // Simulate loading
    const interval = setInterval(() => {
      setValue((prevValue) => prevValue + 1);
    }, 50);

    // Navigate to static host after 2 seconds of loading
    if (value === 100) {
      clearInterval(interval);
      setTimeout(() => {
        turnOn();
      }, 2000);
    }

    return () => clearInterval(interval);
  }, [turnOn, value]);

  return (
    <Box
      bg={'black'}
      display={'flex'}
      flex={1}
      id="powering-container"
      alignItems={'center'}
      height={'100vh'}
      rowGap={100}
      justifyContent={'center'}
      flexDirection={'column'}
    >
      <MacIcon />
      <Box width={250}>
        <Progress
          value={value}
          id="powering-progress"
          borderRadius={10}
          size="sm"
          color={'gray'}
          backgroundColor={'white'}
        />
      </Box>
    </Box>
  );
};

export default Powering;
