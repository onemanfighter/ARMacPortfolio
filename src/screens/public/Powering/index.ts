export { default as PowerOff } from './Powering';
