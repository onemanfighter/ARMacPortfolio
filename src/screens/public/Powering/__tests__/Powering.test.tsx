import { render, screen } from '@testing-library/react';
import Powering from '../Powering';
import { renderHook } from '@testing-library/react-hooks';
import { processStore } from '@processStore';

describe('PowerOff', () => {
  beforeEach(() => {
    const { result } = renderHook(() => processStore());
    result.current.Power.poweringOn();
  });

  it('should render correctly to match snapshot', () => {
    const { container } = render(<Powering />);

    expect(container).toMatchSnapshot();
  });

  it('should render correctly to match snapshot after six second', () => {
    const { container } = render(<Powering />);

    jest.advanceTimersByTime(6000);
    expect(container).toMatchSnapshot();
  });

  it('should show progress bar at 100 once loading time is over', () => {
    render(<Powering />);
    const progress = screen.getByLabelText('powering-progress');
    expect(progress.attributes).toMatchSnapshot();

    // run all timers.
    jest.runAllTimersAsync();
    expect(progress.attributes).toMatchSnapshot();
  });
});
