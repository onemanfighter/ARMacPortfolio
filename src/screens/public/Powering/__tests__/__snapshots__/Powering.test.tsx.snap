// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PowerOff should render correctly to match snapshot 1`] = `
<div>
  <div>
    <h1>
      PowerOff
    </h1>
  </div>
</div>
`;
